### ISSUE TITLE
Single Instance

### ISSUE THREAD

**Issue by funcshen** [nomacs/nomacs#67]
Is this possible? I can't seem to find a setting for it in nomacs or settings.nfo

Win7 x64
3.2 portable

**Comment by diemmarkus** [comment]
nope not yet - sorry

**Comment by Ben-Kaniobi** [comment]
I'd also like to have this option.

It would also be nice to have an additional option for opening a seperate instance only when the new image is not in the same directory as the currently displayed image(s).

**Comment by dreadbalaur** [comment]
So this has been a suggestion since 2016 and it still has not been implemented? Am I understanding correctly that this enhancement is just not achievable in nomacs?

**Comment by scrubbbbs** [comment]
I noticed this was started but never finished. See `DkRunGuard`. What remains is separating the first-launch code (settings loading etc) from nth-launch code (cmdline parsing). Also this will break the sync feature unless there are ways to bypass it.

**Comment by STrRedWolf** [comment]
So... when will this be implemented?  I went looking for a tabbed image viewer and now I'm stuck using Firefox because this doesn't do the job.

**Comment by scrubbbbs** [comment]
We seem too be (at least I am) focused on bugs over features, the backlog is quite large and isn't shrinking yet. Honestly something like this may never happen.

**Comment by MathieuBerthe** [comment]
Hi,
I just wanted to add that it would be amazing if nomacs had this option.

**Comment by dreadbalaur** [comment]
For anyone looking for solutions... I agree that it is unlikely that nomacs will get feature upgrades any time soon. In the meantime I have been using Groupy to circumvent the issue.

**Comment by scrubbbbs** [comment]
I think this can happen at some point, I think we need to fix issues with tabs first as single-instance makes that much more useful.

**Comment by kleineluka** [comment]
Is this something still on the roadmap?

**Comment by scrubbbbs** [comment]
Yes, it is being worked on currently in PR #1633. All of the bugs with tabs are making this take quite a while, and I feel it is not worth releasing if tabs are too buggy.

**Comment by scrubbbbs** [comment]
PR for single-instance is progressing nicely. There are now continous builds in [AppVeyor](https://ci.appveyor.com/project/novomesk/nomacs/history) and github actions, look for the ones tagged "single-instance" if anyone wants to test. That would be helpful as a lot of changes to tabs were required.
